packages/next/server/config.ts

```diff
@@ -126,6 +126,32 @@ function setFontLoaderDefaults(config: NextConfigComplete) {
   } catch {}
 }
 
+function warnOptionHasBeenMovedOutOfExperimental(
+  config: NextConfig,
+  oldKey: string,
+  newKey: string,
+  configFileName: string
+) {
+  if (config.experimental && oldKey in config.experimental) {
+    Log.warn(
+      `\`${oldKey}\` has been moved out of \`experimental\`` +
+        (newKey.includes('.') ? ` and into \`${newKey}\`` : '') +
+        `. Please update your ${configFileName} file accordingly.`
+    )
+
+    let current = config
+    const newKeys = newKey.split('.')
+    while (newKeys.length > 1) {
```

Comment:
Instead of `while` and `newKeys.shift()`, could you use `for (const key of newKeys)`?

Reply:
@styfle I was considering using `for`, but it can't stop before the last key.

Take `compiler.relay` (`['compiler', 'relay']`) as an example, we will need to stop at `['relay']` as only `compiler` requires `current.compiler ||= {}`, and the last key `relay` requires `current.relay = config.experimental.relay`.

That's why I use `while (.length > 1)` here.